File: src/com/vitorblog/mineshop/model/event/PlayerBuyOnShopEvent.java
package com.vitorblog.mineshop.model.event;

import com.vitorblog.mineshop.model.shop.ShopItem;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public final class PlayerBuyOnShopEvent extends Event {
    private static final HandlerList handlers = new HandlerList();
    private Player player;
    private ShopItem shopItem;

    public PlayerBuyOnShopEvent(Player player, ShopItem shopItem) {
        this.player = player;
        this.shopItem = shopItem;
    }

    public Player getPlayer() {
        return player;
    }

    public ShopItem getShopItem() {
        return shopItem;
    }

    public HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: src/com/vitorblog/mineshop/model/event/PlayerBuyOnShopEvent.kt
package com.vitorblog.mineshop.model.event

import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class PlayerBuyOnShopEvent(val player:Player, val item:String) : Event() {

    private val handlers = HandlerList()

    override fun getHandlers(): HandlerList {
        return handlers
    }

    fun getHandlerList(): HandlerList {
        return handlers
    }

}